### PR TITLE
OCT-98: add an user-group option to the akeneo:connectivity-connection:create command

### PR DESCRIPTION
**Pull request by guillaumefer-akeneo**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/back/Application/Settings/Command/CreateConnectionCommand.php [hunk 1]
```diff
@@ -21,18 +21,22 @@ class CreateConnectionCommand
 
     private ?string $type;
 
+    private ?string $userGroup;
```
Comment (RiffVentura) [inline src/Akeneo/Connectivity/Connection/back/Application/Settings/Command/CreateConnectionCommand.php:24]: You could "modernize" DI :wink:

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml [hunk 2]
```diff
@@ -318,6 +318,9 @@ akeneo_connectivity.connection:
             type:
                 required: Connection type is required.
                 too_long: Connection type is too long. It should have 30 characters or less.
+            user_group:
+                must_exist: The user group was not found. Please try again.
```
Comment (RiffVentura) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml:322]:
```suggestion
                must_exist: The user group was not found. Make sure the specified user group exists
```
:thinking:  ?

File: src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Application/Settings/Command/CreateConnectionHandlerSpec.php [hunk 3]
```diff
@@ -86,7 +86,7 @@ public function it_returns_a_connection_with_credentials(
         $createClient->execute('Magento Connector')->shouldBeCalled()->willReturn($client);
 
         $user = new User(42, 'magento_app', 'my_client_pwd');
-        $createUser->execute(Argument::type('string'), 'Magento Connector', ' ')->willReturn($user);
+        $createUser->execute(Argument::type('string'), 'Magento Connector', ' ', null)->willReturn($user);
```
Comment (RiffVentura) [inline src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Application/Settings/Command/CreateConnectionHandlerSpec.php:89]: Missing test case for the user group

**Review by RiffVentura — APPROVED** [review]

File: src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Application/Settings/Command/CreateConnectionHandlerSpec.php (hunk 3 above)
Comment (guillaumefer-akeneo) [reply src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Application/Settings/Command/CreateConnectionHandlerSpec.php]: I replaced `null` by `['All']`. Is it ok for you ?

**Review by masacc — APPROVED** [review]

File: src/Akeneo/Connectivity/Connection/back/tests/.php_cd.php [hunk 4]
```diff
@@ -244,14 +245,15 @@
...
+            'Akeneo\UserManagement\Component\Repository\GroupRepositoryInterface',
 
             // Exceptions
             'Symfony\Component\Validator\Validator\ValidatorInterface',
             'Symfony\Component\Validator\Constraint',
             'Symfony\Component\Validator\ConstraintValidator',
             'Symfony\Component\Validator\Exception\UnexpectedTypeException',
         ]
-    )->in('Akeneo\Connectivity\Connection\Application\Settings'),
+    )->in('Akeneo\Connectiettings/Command/CreateConnectionHandler.php vity\Connection\Application\Settings'),
```
Comment (tseho) [inline src/Akeneo/Connectivity/Connection/back/tests/.php_cd.php:256]: wtf

**Pull request merged**